openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -69,6 +69,7 @@
 public class OpenAPIComponentMapper {
     private final Components components;
     private final List<OpenAPIConverterDiagnostic> diagnostics;
+    private final List<String> cyclicReferences = new ArrayList<>();
```

Comment:
shouldn't this ideally be `visitedTypeDefinitionNames` (or something similar), as here we basically keep track of what's already visited?

```suggestion
    private final List<String> visitedTypeDefinitionNames = new ArrayList<>();
```